webapi/src/main/java/com/dumbdogdiner/stickyapi/webapi/mojang/MojangAPI.java

```diff
@@ -0,0 +1,182 @@
+     * @return A {@link SortedMap} of the time of each change (With 1970-01-01T00:00:00Z meaning that it was the first Username)
+     * @throws HttpException if there is an error with the HTTP Request/Response
+     */
+    public static @NotNull SortedMap<Instant, String> getUsernameHistory(UUID uniqueId) throws HttpException {
+        return getCachedResponse(uniqueId).getUsernameHistory();
+
+    }
+
+    public static byte[] getTexture(@NotNull UUID uniqueId) throws HttpException {
+        return getCachedResponse(uniqueId).getTextures().getSkin().getData();
+    }
+
+    public static @Nullable String getTextureString(@NotNull UUID uniqueId) throws HttpException {
+        return getCachedResponse(uniqueId).getTextureString();
+    }
+
+    public static @NotNull String getUsername(UUID uniqueId) throws HttpException {
+        return getCachedResponse(uniqueId).getUsername();
+    }
+
+    public static @NotNull UUID getUniqueId(String username) throws HttpException {
+        HttpUrl url = PLAYERDB_MINECRAFT_URL.resolve(username);
+        JsonElement response = HttpUtil.getResponseAsJson(url);
+        try {
+            return UUID.fromString(
+                    response.getAsJsonObject()
+                            .get("data").getAsJsonObject()
+                            .get("player").getAsJsonObject()
+                            .get("meta").getAsJsonObject()
+                            .get("id").getAsString());
+        } catch (Exception e) {
+            throw new HttpException(url, "Invalid JSON received");
+        }
+    }
+
+    public static @NotNull UUID getUniqueId(@NotNull String username, @NotNull Instant when) throws HttpException {
+        return uuidAtTime(username, when);
+    }
+
+    public static @NotNull Map<String, MojangStatus> getMojangAPIStatus() throws HttpException {
+        @NotNull Map<String, MojangStatus> status = new HashMap<>();
+
+        for (@NotNull JsonElement obj : HttpUtil.getResponseAsJson(MOJANG_STATUS_URL).getAsJsonArray()) {
+            for (Map.@NotNull Entry<String, JsonElement> entry : obj.getAsJsonObject().entrySet()) {
+                status.put(entry.getKey(), MojangStatus.valueOf(entry.getValue().getAsString().toUpperCase()));
+            }
+        }
+
+        return status;
+    }
+
+    /**
+     * Get the created time of an account. Presently sorta broken, if the time is not available it will return an instant of -1
+     *
+     * @param uniqueId The UUID to get the created time for
+     * @return The created time as an instant if possible, or an Instant representing -1 Seconds Unix time if not
+     */
+    public static @NotNull Instant getCreated(UUID uniqueId) throws HttpException {
+        return getCachedResponse(uniqueId).getCreated();
+    }
+
+    /**
+     * Provides the UUID associated with a given username at a given time.
+     * FIXME: This is currently broken, with an <a href="https://bugs.mojang.com/browse/WEB-3367>open bug report at Mojang</a>
```

Comment:
Looks like JavaDoc might complain about this.
```suggestion
     * FIXME: This is currently broken, with an <a href="https://bugs.mojang.com/browse/WEB-3367">open bug report at Mojang</a>
```

Reply:
Fixed

Reply:
Looks broken on my end still. Not included in commit perhaps?

Reply:
I'm just marking them off as I go so I'm not a dumbwoof

Reply:
Because I've been asked to make less commits

Reply:
Who told you that? Many commits are fine as long as you label them properly :3